### DISCUSSION TITLE
OKD4.7 offline installation(openstack)

### DISCUSSION THREAD

**Discussion by Mohammed-Idrees** [okd-project/okd#866] [Q&A]
Hi Team,
I'm trying to install OKD 4.7 (4.7.0-0.okd-2021-05-22-050008) using mirror registry for offline installation(no internet), I'm able to see bootstrap node is able to pull the images from the mirror registry and when I execute "openshift-install --dir=install_dir/ wait-for bootstrap-complete --log-level=debug" command I'm able to see API is up but the master nodes are not able to pull the images from the mirror registry and I'm not able to ssh to any of the master nodes. is it possible to install OKD4.7 on a cluster without internet?

![image](https://user-images.githubusercontent.com/36164488/132165213-820b6105-d736-4252-98c7-053533d15c86.png)

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

>the master nodes are not able to pull the images from the mirror registry

On this screen master is unable to provision at all

**Comment by Mohammed-Idrees** [comment]
Hi @vrutkovs 

Please find the attached log bundle.
[log-bundle-20210906113141.tar.gz](https://github.com/openshift/okd/files/7115737/log-bundle-20210906113141.tar.gz)

**Comment by Mohammed-Idrees** [comment]
Hi @vrutkovs 

Is this a bug/ known issue? is there any workaround for this?
can you please let me know any solution on this.

**Comment by vrutkovs** [comment]
No data collected from masters and apparently these never requested their Ignition.

Check LB configuration and network settings for masters

**Comment by dfateyev** [comment]
FYI, I have recently stumbled upon this issue several times, and worked around using `okd-2021-07-03-190901` and `fedora-coreos-34.20210626.3.2` where I it's not observed. Not sure though whether it's an issue with OKD or FCOS, didn't investigate it further.

**Reply by vrutkovs** [reply]
See https://github.com/openshift/okd/issues/783 - FCOS issue we worked around in OKD (by defaulting to older FCOS)

**Comment by Mohammed-Idrees** [comment]
Hi Team,
I'm trying to install OKD 4.8 (4.8.0-0.okd-2021-10-24-061736) using mirror registry for offline installation(no internet), I'm able to get the mirror repo from the container and get the image list as well.
curl -u admin:admin -k https://mirror-registry:5000/v2/_catalog
{"repositories":["openshift/okd"]}

I have enabled the required ports for the okd. I have tried using the FCOS image version 34.20210626.3.1  as suggested in [#783](https://github.com/openshift/okd/issues/783) and tried using FCOS version 34.20210919.3.0 , But couldn't get the work done, it also results in the dial tcp i/o timeout error.

If the internet is enabled I'm able to create OKD 4.8 cluster with the FCOS version 34.20210919.3.0 and OC version 4.8.0-0.okd-2021-10-24-061736.

I'm able to see bootstrap node is able to pull the images from the mirror registry and when I execute "openshift-install --dir=install_dir/ wait-for bootstrap-complete --log-level=debug" command I'm able to see API is up but the master nodes are not able to pull the images from the mirror registry and I'm not able to ssh to any of the master nodes.

Is there any workaround/procedure for OKD4.8 offline installation.? 
[open-port-list.txt](https://github.com/openshift/okd/files/7660287/open-port-list.txt)

![image](https://user-images.githubusercontent.com/36164488/144836100-4124f0f7-0623-4911-86f3-1db427142bda.png)

**Reply by LorbusChris** [reply]
Are the nodes able to talk to each other? What is your DNS setup? Do nodes get their names from DHCP via a reverse DNS lookup, or is there proper DNS configured? It looks to me as though you're using a `local` domain which points towards mDNS which I don't think is supported.

**Reply by Mohammed-Idrees** [reply]
Hi @LorbusChris ,
bootstrap node able to talk to the service node and master node (using IPs).

DNS setup is as per these [okd4 files](https://github.com/cragr/okd4_files).

As I'm using OpenStack VMs(DHCP is configured on OpenStack) nodes get their names like host-192-167-27-100 (assuming 192.168.27.100 is the IP).

dig commands show proper results.(dig okd.local and dig -x <service-ip>).
